deployment.go

```diff
@@ -56,11 +56,51 @@ type StepActions struct {
 
 // DeploymentPlan is a collection of steps for application deployment
 type DeploymentPlan struct {
-	ID       string            `json:"id"`
-	Version  string            `json:"version"`
-	Original *Group            `json:"original"`
-	Target   *Group            `json:"target"`
-	Steps    []*DeploymentStep `json:"steps"`
+	ID       	string            	`json:"id"`
+	Version  	string            	`json:"version"`
+	Original 	*Group            	`json:"original"`
+	Target   	*Group            	`json:"target"`
+	Steps    	[]*DeploymentStep 	`json:"-"`
+	XXStepsRaw	json.RawMessage		`json:"steps"` // Holds raw steps JSON to unmarshal later
+}
+
+// Allows loading of deployment steps from the Marathon v1.X API
+func (d *DeploymentPlan) UnmarshalJSON(b []byte) error {
+	dp := &struct {
+		ID       	string            	`json:"id"`
+		Version  	string            	`json:"version"`
+		Original 	*Group            	`json:"original"`
+		Target   	*Group            	`json:"target"`
+		XXStepsRaw  json.RawMessage 	`json:"steps"`
+	}{}
+	err := json.Unmarshal(b, &dp)
+	if err != nil {
+		return err
+	}
+	d.ID = dp.ID
+	d.Version = dp.Version
+	d.Original = dp.Original
+	d.Target = dp.Target
+	var deploymentStepArray []*DeploymentStep
+	if len(dp.XXStepsRaw) != 0 {
```

Comment:
There's possibly some overlap with the [implementation of Deployments()](https://github.com/gambol99/go-marathon/blob/8908223430b37fc4106d8b8cf1f38cf7124020f1/deployment.go#L67). Would be interesting to see if we can extract the logic into a common place if that doesn't make it more complicated.